### PR TITLE
fix(Dropdown): Prevented closing if menu is in multiple mode

### LINKED ISSUE #45520: Dropdown closing if menu is in multiple mode
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-forked-6662cn)

### Steps to reproduce

Pass in \<Dropdown\> menu prop with items, selectable: true, multiple: true

### What is expected?

Dropdown is still open after select and deselect an item

### What is actually happening?

Dropdown always closes

| Environment | Info |
| --- | --- |
| antd | 5.10.2 |
| React | 18.2.0 |
| System | macOS 13.2 |
| Browser | Chrome 118.0.5993.117 |

### PR DESCRIPTION
**Pull request by vyachsed**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix [#45520](https://github.com/ant-design/ant-design/issues/45520)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |Fix Dropdown always closes after clicking on an item|
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at d3240af</samp>

Fixed a bug in `dropdown.tsx` that caused the dropdown menu to close when selecting multiple options. Added `menu` prop as a dependency to `onMenuClick` function.

### <samp>🤖 Generated by Copilot at d3240af</samp>

*  Prevent closing dropdown menu when multiple selectable ([link](https://github.com/ant-design/ant-design/pull/45513/files?diff=unified&w=0#diff-978b29ecd5429963a073f5adfa7ded79d88d23850853c9fb6d453a82047365beL216-R220))
*  Add `menu` prop as dependency to `onMenuClick` function ([link](https://github.com/ant-design/ant-design/pull/45513/files?diff=unified&w=0#diff-978b29ecd5429963a073f5adfa7ded79d88d23850853c9fb6d453a82047365beL216-R220))

### CODE REVIEW

**Comment by afc163** [comment]
Could you add a test case?

File: components/dropdown/dropdown.tsx [hunk 1]
```diff
@@ -213,8 +213,11 @@ const Dropdown: CompoundedComponent = (props) => {
   });
 
   const onMenuClick = React.useCallback(() => {
+    if (menu?.selectable && menu?.multiple) {
+      return;
+    }
     setOpen(false);
-  }, []);
+  }, [menu]);
```
Comment (afc163) [inline components/dropdown/dropdown.tsx:220]:
```suggestion
  }, [menu?.selectable, menu?.multiple]);
```

**Comment by vyachsed** [comment]
> Could you add a test case?

Do you mean formalized issue (https://github.com/ant-design/ant-design/issues/45520)? Or should I add some tests for my changes?

**Comment by afc163** [comment]
Right, the original problem should be covered.

**Comment by vyachsed** [comment]
> #45520)

Sorry, didn't catch you.
Do you mean that the problem should be described? (It's done, I attached the issue link)
Or do you mean that I should cover my code with tests?

**Comment by afc163** [comment]
We should add test case (such as https://github.com/ant-design/ant-design/blob/master/components/avatar/__tests__/Avatar.test.tsx ) for your changes, in case the original problem happen again.

**Comment by vyachsed** [comment]
> We should add test case (such as https://github.com/ant-design/ant-design/blob/master/components/avatar/__tests__/Avatar.test.tsx ) for your changes, in case that the original problem happen again.

Done

**Review by MadCcc — APPROVED** [review]

**Comment by kienntt** [comment]
When i upgrade to version  5.11.3 it still closes after clicking on an item, then i comback version 5.10.3

**Pull request merged**